src/usr/local/www/diag_ndp.php

```diff
@@ -33,10 +33,28 @@
 
 @ini_set('zlib.output_compression', 0);
 @ini_set('implicit_flush', 1);
-
+define('NDP_BINARY_PATH', '/usr/sbin/ndp');
 require_once("guiconfig.inc");
 
-exec("/usr/sbin/ndp -na", $rawdata);
+// Delete ndp entry.
+if (isset($_POST['deleteentry'])) {
+	$ip = $_POST['deleteentry'];
+	if (is_ipaddrv6($ip)) {
+		$commandReturnValue = mwexec(NDP_BINARY_PATH . " -d " . $ip, true);
```

Comment:
The IPv6 address value must be validated and escaped. If it's not a valid IPv6 address, the command should not be run at all. And even when it is, the value must be escaped using `escapeshellarg()`.